source/extensions/filters/http/ext_proc/mutation_utils.cc

```diff
@@ -174,40 +182,132 @@ absl::Status MutationUtils::applyHeaderMutations(const HeaderMutation& mutation,
       return absl::InvalidArgumentError("Invalid character in set_headers mutation.");
     }
     const LowerCaseString header_name(sh.header().key());
-    const bool append = PROTOBUF_GET_WRAPPED_OR_DEFAULT(sh, append, false);
-    const auto check_op = (append && !headers.get(header_name).empty()) ? CheckOperation::APPEND
-                                                                        : CheckOperation::SET;
-    auto check_result = checker.check(check_op, header_name, header_value);
-    if (replacing_message && header_name == Http::Headers::get().Method) {
-      // Special handling to allow changing ":method" when the
-      // CONTINUE_AND_REPLACE option is selected, to stay compatible.
-      check_result = CheckResult::OK;
-    }
-    switch (check_result) {
-    case CheckResult::OK:
-      ENVOY_LOG(trace, "Setting header {} append = {}", sh.header().key(), append);
-      if (append) {
-        headers.addCopy(header_name, header_value);
-      } else {
-        headers.setCopy(header_name, header_value);
+
+    if (Runtime::runtimeFeatureEnabled(
+            "envoy.reloadable_features.header_value_option_change_action")) {
```

Comment:
So, you can do:
if (sh.append_action().has_value()) ...

Reply:
Thanks, I will test one more time on this approach, I will let you know how it goes.

Reply:
I have updated the code according to your suggestion.
Unfortunately it failed.

source/extensions/filters/http/ext_proc/mutation_utils.cc:186:27: error: member reference base type '::envoy::config::core::v3::HeaderValueOption_HeaderAppendAction' is not a structure or union
    if (sh.append_action().has_value()) {
        ~~~~~~~~~~~~~~~~~~^~~~~~~~~~
Detailed error can be located at [here](https://dev.azure.com/cncf/envoy/_build/results?buildId=152270&view=logs&j=767be981-567e-57d8-68c3-2140ede0a0bd&t=67a76341-37f4-5b49-db38-acbb8a59af82&l=611)